view/frontend/web/js/view/payment/method-renderer/hosted-view-payment.js

```diff
@@ -0,0 +1,222 @@
+define([
+    'Magento_Checkout/js/view/payment/default',
+    'ko',
+    'jquery',
+    'mage/storage',
+    'Magento_Checkout/js/action/place-order',
+    'Magento_Checkout/js/action/select-payment-method',
+    'Magento_Checkout/js/model/quote',
+    'Magento_Customer/js/model/customer',
+    'Magento_Checkout/js/model/payment-service',
+    'Magento_Checkout/js/checkout-data',
+    'Magento_Checkout/js/model/checkout-data-resolver',
+    'uiRegistry',
+    'Magento_Checkout/js/model/payment/additional-validators',
+    'Magento_Ui/js/model/messages',
+    'Magento_Ui/js/model/messageList',
+    'uiLayout',
+    'Magento_Checkout/js/action/redirect-on-success',
+    'Magento_Checkout/js/model/full-screen-loader',
+    'Magento_Customer/js/customer-data',
+    'mage/url',
+    'mage/cookies',
+], function (
+    Component,
+    ko,
+    $,
+    storage,
+    placeOrderAction,
+    selectPaymentMethodAction,
+    quote,
+    customer,
+    paymentService,
+    checkoutData,
+    checkoutDataResolver,
+    registry,
+    additionalValidators,
+    Messages,
+    messageContainer,
+    layout,
+    redirectOnSuccessAction,
+    fullscreenLoader,
+    customerData,
+    urlBuilder,
+    cookies
+) {
+    'use strict';
+
+    const hostedUrl = ko.observable('');
+
+    return Component.extend({
+        defaults: {
+            template: 'SwedbankPay_Payments/payment/hosted-view'
+        },
+        config: {
+            data: {
+                culture: 'en-US',
+                logo: 'SwedbankPay_Payments/images/swedbank-pay-logo.svg'
+            }
+        },
+        hostedUrl: hostedUrl,
+        logoUrl: function () {
+            return require.toUrl(this.config.data.logo);
+        },
+        initialize: function () {
+            let self = this;
+            self.totals = {};
+            self.instrumentScript = '';
+
+            self._super();
+            Object.assign(this.config.data, window.checkoutConfig.SwedbankPay_Payments);
+            Object.assign(this.config.data, window.checkoutConfig.SwedbankPay_Payments_Instrument_List);
+
+            // quote.totals.subscribe(function(totals) {
+            //     if (self.totals.grand_total !== totals.grand_total) {
+            //         if (self.getCode() == self.isChecked()) {
+            //             self.updatePaymentScript();
+            //         }
+            //     }
+            //
+            //     self.totals = totals;
+            // });
+        },
+        clearPaymentScript: function() {
+            let self = this;
+
+            self.getAvailableInstruments().forEach(function (instrument) {
+                if (typeof payex !== "undefined" && typeof payex.hostedView[instrument.js_object_name] !== "undefined") {
+                    payex.hostedView[instrument.js_object_name]().close();
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: ['payex' is not defined.](https://app.codacy.com/gh/SwedbankPay/swedbank-pay-magento2-payments/pullRequest?prid=5056328)